pandas/tests/frame/test_analytics.py

```diff
@@ -138,6 +138,18 @@ def test_corr_invalid_method(self):
         with tm.assert_raises_regex(ValueError, msg):
             df.corr(method="____")
 
+    def test_corr_tril(self):
```

Comment:
For these tests, it's better to build an expected DataFrame output and compare it to a resulting DataFrame. You can build a DataFrame that gives a trivial correlation matrix (like a `np.ones` DataFrame)

```
result = df.corr(tri=...)
expected = pd.DataFrame(...)
tm.assert_frame_equal(result, expected)

```

Reply:
I updated the tests according to this idea.  We're still using a `DataFrame` inside `ones_like`, but I figured this is okay since they're small test cases.